src/Merge/ExtraPackage.php

```diff
@@ -165,6 +171,20 @@ public function mergeInto(RootPackageInterface $root, PluginState $state)
         }
 
         $this->mergeExtra($root, $state);
+
+        // Merge source reference information for merged packages.
+        // @see RootPackageLoader::load
+        $references = array();
+        foreach (array('require', 'require-dev') as $linkType) {
+            $linkInfo = BasePackage::$supportedLinkTypes[$linkType];
+            $method = 'get'.ucfirst($linkInfo['method']);
+            $links = array();
+            foreach ($root->$method() as $link) {
+                $links[$link->getTarget()] = $link->getConstraint()->getPrettyString();
+            }
+            $references = $this->extractReferences($links, $references);
+        }
+        $root->setReferences($references);
```

Comment:
@bd808 setReferences is not part of the ``RootPackageInteface``. Should we use ``unwrapIfNeeded`` or should we try to fix it upstream?